File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    #원리는 알았으나 코드 표현력이 부족했음
    
    for i in arr:
        if not answer:
            answer.append(i)
        elif answer and answer[-1] == i:
            answer.pop()
            answer.append(i)
        else:
            answer.append(i)
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        Stack<Integer> stack = new Stack<Integer>();

        for (int i = 0; i < arr.length; i++) {
            if (stack.isEmpty() || stack.peek() != arr[i]) {
                stack.push(arr[i]);
            }
        }

        for (Integer integer : stack) {
            answer.add(integer);
        }

        return answer.stream()
                .mapToInt(Integer::intValue).toArray();
    }
}